examples/bridge-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,452 @@
+    DECLARE_DYNAMIC_CLUSTER(ZCL_BRIDGED_DEVICE_BASIC_CLUSTER_ID, bridgedDeviceBasicAttrs),
+    DECLARE_DYNAMIC_CLUSTER(ZCL_FIXED_LABEL_CLUSTER_ID, fixedLabelAttrs) DECLARE_DYNAMIC_CLUSTER_LIST_END;
+
+// Declare Bridged Light endpoint
+DECLARE_DYNAMIC_ENDPOINT(bridgedLightEndpoint, bridgedLightClusters);
+
+int AddDeviceEndpoint(Device * dev, EmberAfEndpointType * ep, uint16_t deviceType)
+{
+    uint8_t index = 0;
+    while (index < DYNAMIC_ENDPOINT_COUNT)
+    {
+        if (NULL == gDevices[index])
+        {
+            gDevices[index] = dev;
+            EmberAfStatus ret;
+            while (1)
+            {
+                ret = emberAfSetDynamicEndpoint(index, gCurrentEndpointId, ep, deviceType, DEVICE_VERSION_DEFAULT);
+                if (ret == EMBER_ZCL_STATUS_SUCCESS)
+                {
+                    ChipLogProgress(DeviceLayer, "Added device %s to dynamic endpoint %d (index=%d)", dev->GetName(),
+                                    gCurrentEndpointId, index);
+                    return index;
+                }
+                else if (ret != EMBER_ZCL_STATUS_DUPLICATE_EXISTS)
+                {
+                    return -1;
+                }
+                // Handle wrap condition
+                if (++gCurrentEndpointId < gFirstDynamicEndpointId)
+                {
+                    gCurrentEndpointId = gFirstDynamicEndpointId;
+                }
+            }
+        }
+        index++;
+    }
+    ChipLogProgress(DeviceLayer, "Failed to add dynamic endpoint: No endpoints available!");
+    return -1;
+}
+
+int RemoveDeviceEndpoint(Device * dev)
+{
+    uint8_t index = 0;
+    while (index < DYNAMIC_ENDPOINT_COUNT)
+    {
+        if (gDevices[index] == dev)
+        {
+            EndpointId ep   = emberAfClearDynamicEndpoint(index);
+            gDevices[index] = NULL;
+            ChipLogProgress(DeviceLayer, "Removed device %s from dynamic endpoint %d (index=%d)", dev->GetName(), ep, index);
+            // Silence complaints about unused ep when progress logging
+            // disabled.
+            UNUSED_VAR(ep);
+            return index;
+        }
+        index++;
+    }
+    return -1;
+}
+
+// ZCL format -> (len, string)
+uint8_t * ToZclCharString(uint8_t * zclString, const char * cString, uint8_t maxLength)
+{
+    size_t len = strlen(cString);
+    if (len > maxLength)
+    {
+        len = maxLength;
+    }
+    zclString[0] = static_cast<uint8_t>(len);
+    memcpy(&zclString[1], cString, zclString[0]);
+    return zclString;
+}
+
+// Converted into bytes and mapped the (label, value)
+void EncodeFixedLabel(const char * label, const char * value, uint8_t * buffer, uint16_t length, EmberAfAttributeMetadata * am)
+{
+    char zclOctetStrBuf[kFixedLabelElementsOctetStringSize];
+    uint16_t listCount = 1;
+    _LabelStruct labelStruct;
+
+    labelStruct.label = chip::ByteSpan(reinterpret_cast<const uint8_t *>(label), kFixedLabelElementsOctetStringSize);
```

Comment:
This does not make any sense.  Callers pass things like "room", but here we are going to read `kFixedLabelElementsOctetStringSize == 16` bytes from that pointer?  That's going to read random memory...  

I really wish we were adding tests to go along with this code....

Reply:
Yeah. Corrected.